docs/advanced-topics/serialization/inetworkserializable.md

```diff
@@ -73,32 +72,114 @@ The below `INetworkSerializable` implementation example works only with RPCs and
 [Read More About Custom NetworkVariable Implementations](../../basics/networkvariable.md)
 :::
 
+Because of the limitations of NetworkVariable, which is implemented to avoid creating garbage as much as possible, C# arrays aren't supported in NetworkVariable, including arrays that are embedded in structs, as they prevent those structs from being allocated without creating garbage. Unity provides alternatives to C# arrays, however, that do work with NetworkVariable - `NativeArray<>` and `NativeList<>`.
```

Comment:
The `caution` section just above all this new text is now incorrect (it says the below example won't work with NetVar because of array usage).